### DISCUSSION TITLE
How should extensions using global `fetch` adapt to the new `http.fetchAdditionalSupport` setting?

### DISCUSSION THREAD

**Discussion by shouples** [microsoft/vscode-discussions#2438] [Extension Development QnA] [answered]
With the 1.96 release and the new [`http.fetchAdditionalSupport`](https://code.visualstudio.com/updates/v1_96#_proxy-support-for-nodejs-fetch-api) setting, we noticed that leaving it **_enabled_** would cause issues with our extension making `fetch` requests to the local Docker engine API's `/_ping` endpoint when using a custom `dispatcher` to specify the socket path, e.g.:
```ts
import { Agent } from "undici";
// ...
const init = {
  dispatcher: new Agent({
    connect: {
      socketPath: "/var/run/docker.sock",
    },
  })
}
```

Sample error info:
```
TypeError: fetch failed
    at node:internal/deps/undici/undici:13392:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at globalThis.fetch (file:///Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:159:19800) 
...
cause:
ECONNREFUSED: fetch failed
AggregateError
    at internalConnectMultiple (node:net:1122:18)
    at afterConnectMultiple (node:net:1689:7)
    at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17)
```
(I'm not sure what the best format is for this kind of error, but I'm happy to provide more information if needed.)

With the `http.fetchAdditionalSupport` setting **_disabled_**, we see successful responses returned from that endpoint as expected. Other `fetch` requests do not appear to be having any issues as far as we've seen.

Is there a recommended path for extension developers to work with/around this setting?

**Comment by shouples** [comment]
Update after looking at the related settings:
![image](https://github.com/user-attachments/assets/a5e1b2cf-6b03-493b-a554-c83c84d7ca53)

If I toggle the `http.systemCertificates` setting, the requests described above are affected:
- `http.fetchAdditionalSupport` enabled and `http.systemCertificates` enabled: requests fail with the above error ❌
- `http.fetchAdditionalSupport` enabled and `http.systemCertificates` disabled: requests return successful responses ✅ 

(Toggling `http.proxySupport` seems to have no effect here.)

Should this be a bug issue in https://github.com/microsoft/vscode?

**Reply by olguzzar** [reply]
Hi @shouples, yes go ahead and file an issue on that repo since this change occurred after the recent update. Thanks!

**Reply by shouples** [reply, marked as answer]
Thanks @olguzzar - issue created https://github.com/microsoft/vscode/issues/236423
